plugins/dell-k2/fu-dell-k2-plugin.c

```diff
@@ -415,6 +436,9 @@ fu_dell_k2_plugin_constructed(GObject *obj)
 	fu_plugin_add_firmware_gtype(plugin, NULL, FU_TYPE_DELL_K2_RTSHUB_FIRMWARE);
 	fu_plugin_add_firmware_gtype(plugin, NULL, FU_TYPE_DELL_K2_DPMUX_FIRMWARE);
 	fu_plugin_add_firmware_gtype(plugin, NULL, FU_TYPE_DELL_K2_ILAN_FIRMWARE);
+
+	/* defaults changed here will also be reflected in the fwupd.conf man page */
+	fu_plugin_set_config_default(plugin, FWUPD_DELL_K2_PLUGIN_CONFIG_UOD, "true");
```

Comment:
so you're flipping the default?

Reply:
Users can flip it for an immediate update without needing to unplug the cable.

Reply:
I'd be surprised if 0.01% of users even know there is a config file. The default we choose is hugely important.

Reply:
Yes, the user experience will be consistent with that of the previous docks.

Reply:
So just to double check intent marches code.

Uod is enabled by default and users who want immediate update (which is different than last gen)will change config file to disable.

Right?

Reply:
> Uod is enabled by default and users who want immediate update (which is different than last gen)will change config file to disable.

Exactly, uod i.e. update-on-disconnect should be default enabled. 

On the contrary, users who want to activate the updates immediately after the firmware is installed should modify the plugin configuration before performing an update. So the new version will take effect without needing to unplug the dock cable. 

However, all devices in the dock will be disconnected immediately after the firmware updates are installed.

Reply:
So I think you need to change the value of the flags USABLE_WHILE_UPDATING based on this config value.

Reply:
for all the devices respect to UOD or just EC?